### PR TITLE
🎉Source Facebook Marketing: Updated date validation process

### LINKED ISSUE #15243: Facebook Marketing: Automatically update start_date if it earlier than 37 months in the past
## Tell us about the problem you're trying to solve
The FB Marketing API only allows us to fetch data for the last 37 months. We currently fail the sync if the start date is too early.

## Describe the solution you’d like
Instead of failing the sync if the start date is too early, we set the start_date to the first valid date (`pendulum.now() - DATA_RETENTION_PERIOD`).

## Describe the alternative you’ve considered or used
An alternative is to ask the customer to update their configuration when their start date becomes too far in the past, but that's not a great UX

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
#15243 - Facebook Marketing: Automatically update start_date if it earlier than 37 months in the past

## How
Updated date validation utils and unit tests.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
</details>

### CODE REVIEW

**Review by roman-yermilov-gl — APPROVED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/utils.py [hunk 1]
```diff
@@ -2,35 +2,42 @@
...
+            f"The start date cannot be beyond {DATA_RETENTION_PERIOD} months from the current date. "
+            f"Set start date to {current_date.subtract(months=DATA_RETENTION_PERIOD)}."
+        )
+        logger.warning(message)
+        return current_date.subtract(months=DATA_RETENTION_PERIOD)
+    return start_date
+
+
+def validate_end_date(start_date: datetime, end_date: datetime) -> datetime:
+    if start_date > end_date:
```
Comment (girarda) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/utils.py:39]: should requesting a single day be valid? eg from "2022-08-04" to "2022-08-04"

**Review by girarda — APPROVED** [review]
looks good! just a small question

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/utils.py (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/utils.py:39]: yes, from "2022-08-04" to "2022-08-04" is valid range.

**Pull request merged**